extensions/wikia/Discussions/EnableDiscussionsController.class.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+class EnableDiscussionsController extends \WikiaController {
+
+	const ENABLE_DISCUSSIONS_NAV = 'wgEnableDiscussionNavigation';
+	const ENABLE_DISCUSSIONS = 'wgEnableDiscussion';
+	const ENABLE_FORUMS = 'wgEnableForumExt';
+
+	const P_SITE_ID = 'siteId';
+	const P_ROLLBACK = 'rollback';
+	const REASON = 'Enabling discussions!';
+
+	public function index() {
+		$this->assertCanAccessController();
+
+		$siteId = $this->request->getInt( self::P_SITE_ID );
+		$isRollback = $this->request->getBool( self::P_ROLLBACK );
+
+		if ( $siteId < 1 ) {
+			throw new InvalidArgumentException( 'Please provide a valid siteId, provided ' .
+			                                    $siteId );
+		}
+
+		$discussions = $this->setVariable( $siteId, self::ENABLE_DISCUSSIONS, !$isRollback );
+		$navigation = $this->setVariable( $siteId, self::ENABLE_DISCUSSIONS_NAV, !$isRollback );
+		$forum = $this->setVariable( $siteId, self::ENABLE_FORUMS, $isRollback );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)